verilog/tools/lint/lint_tool_test.sh

```diff
@@ -268,4 +268,213 @@ grep -q "module-parameter" "${MY_OUTPUT_FILE}.err" || {
+interactive_autofix_test() {
+  local keys_="$1"
+  shift
+
+  cp "${ORIGINAL_TEST_FILE}" "${TEST_FILE}"
+
+  "$lint_tool" --ruleset=none --rules_config="${RULES_CONFIG_FILE}" --autofix=interactive \
+      "${TEST_FILE}" > /dev/null 2>&1 <<< "$keys_"
+
+  local IFS=''
+  echo -en "$*" > "${REFERENCE_TEST_FILE}"
+
+  diff -u "${REFERENCE_TEST_FILE}" "${TEST_FILE}" >"${DIFF_FILE}"
+  status="$?"
+  (( $status )) && {
+    echo "Reference file and fixed test file differ:"
+    cat "${DIFF_FILE}"
+    exit 1
+  }
+}
+
+echo "=== Test --autofix=interactive: Apply All"
+
+interactive_autofix_test "A" \
+    'module Autofix;\n' \
+    '  wire a;\n'       \
+    '  wire b;\n'       \
+    '  wire c;\n'       \
+    '  wire d;\n'       \
+    'endmodule\n'
+
+echo "=== Test --autofix=interactive: Reject All"
+
+interactive_autofix_test "D" \
+  'module Autofix;    \n' \
+  '  wire a;;\n'          \
+  '  wire b;;  \n'        \
+  '  wire c;;\n'          \
+  '  wire d;;\n'          \
+  'endmodule    '
+
+echo "=== Test --autofix=interactive: Reject"
+
+interactive_autofix_test "nA" \
+    'module Autofix;    \n' \
+    '  wire a;\n'           \
+    '  wire b;\n'           \
+    '  wire c;\n'           \
+    '  wire d;\n'           \
+    'endmodule\n'
+
+echo "=== Test --autofix=interactive: Apply"
+
+interactive_autofix_test "yD" \
+    'module Autofix;\n'    \
+    '  wire a;;\n'         \
+    '  wire b;;  \n'       \
+    '  wire c;;\n'         \
+    '  wire d;;\n'         \
+    'endmodule    '
+
+echo "=== Test --autofix=interactive: Apply All For Rule"
+
+interactive_autofix_test "annnnn" \
+    'module Autofix;\n'  \
+    '  wire a;;\n'       \
+    '  wire b;;\n'       \
+    '  wire c;;\n'       \
+    '  wire d;;\n'       \
+    'endmodule'
+
+echo "=== Test --autofix=interactive: Reject All For Rule"
+
+interactive_autofix_test "dyyyyy" \
+    'module Autofix;    \n' \
+    '  wire a;\n'           \
+    '  wire b;  \n'         \
+    '  wire c;\n'           \
+    '  wire d;\n'           \
+    'endmodule    \n'
+
+echo "=== Test --autofix=interactive: Reject All For Rule, Apply All For Rule"
+
+interactive_autofix_test "dan" \
+    'module Autofix;    \n' \
+    '  wire a;\n'           \
+    '  wire b;  \n'         \
+    '  wire c;\n'           \
+    '  wire d;\n'           \
+    'endmodule    '
+
+echo "=== Test --autofix=interactive: Show Fix"
+
+interactive_autofix_test "yppnnpypyyppyn" \
+    'module Autofix;\n'  \
+    '  wire a;;\n'       \
+    '  wire b;;\n'       \
+    '  wire c;\n'        \
+    '  wire d;\n'        \
+    'endmodule'
+
+echo "=== Test --autofix=interactive: Show Applied Fixes"
+
+interactive_autofix_test "yPPnnPyPyyPPyn" \
+    'module Autofix;\n'  \
+    '  wire a;;\n' \
+    '  wire b;;\n' \
+    '  wire c;\n'  \
+    '  wire d;\n'  \
+    'endmodule'
+
+################################################################################
+
```

Comment:
Another interesting test might be if the lint tool is supplied multiple filenames, so that we can test that the patch output contains the right filenames to patch.

Reply:
Good idea, turns out there are actually some problems with autofix patch for multiple files. I'll add tests and fix problems.

Reply:
Done